target/riscv/csr.c

```diff
@@ -30,7 +30,7 @@ static riscv_csr_operations csr_ops[];
 /* CSR function table constants */
 
 enum {
-    CSR_TABLE_SIZE = 0xfff
+    CSR_TABLE_SIZE = 0x1000
```

Comment:
Good spotting. I vaguely remember this one. CSR 0xfff is out of bounds. We should roll this fragment into https://github.com/riscv/riscv-qemu/commit/0bf57161b8c41dda478b9ce4ed6954ecd8ba0476 _RISC-V: Implement modular CSR helper interface_ When I merge the commit I can rebase and move this fragment.

Reply:
With an earlier version of the patch, I got a core dump when gdb tried to read CSR4095, because of the out-of-bounds array access.